include/nuttx/tls.h

```diff
@@ -309,9 +315,28 @@ uintptr_t task_tls_get_value(int tlsindex);
 #elif defined(CONFIG_TLS_ALIGNED) && !defined(__KERNEL__)
```

Comment:
Does someone know what up_tls_info is about ? I can not find any references for it in the project ?

Reply:
up_tls_info is for the grow up stack which doesn't exist on all supported arch.

Reply:
I mean I can not find a single reference in the upstream nuttx project, do you think such downstream implementations can exist ?

```
ville@workpc:~/wpc_toolchain/workspace/nuttx-bare/nuttx$ grep up_tls_info -nr .
./libs/libc/tls/tls_getinfo.c:32:#if !defined(up_tls_info) && (defined(__KERNEL__) || !defined(CONFIG_TLS_ALIGNED))
./libs/libc/tls/tls_getinfo.c:74:#endif /* !defined(up_tls_info) && (defined(__KERNEL__) || !defined(CONFIG_TLS_ALIGNED)) */
./include/nuttx/tls.h:313:#if defined(up_tls_info)
./include/nuttx/tls.h:314:#  define tls_get_info() up_tls_info()
./include/nuttx/arch.h:1935: * Name: up_tls_info
./include/nuttx/arch.h:1963: * FAR struct tls_info_s *up_tls_info(void);
./Documentation/ReleaseNotes/NuttX-10.3.0:100:* [#5629](https://github.com/apache/nuttx/pull/5629) libc: fix up_tls_info define for BUILD_KERNEL 
./Documentation/ReleaseNotes/NuttX-10.3.0:231:* [#4941](https://github.com/apache/nuttx/pull/4941) arch: Remove the duplicated up_tls_info implementation 
```

Reply:
no, as far as I know.

Reply:
The thing I'm not sure about is how to implement nxnotify_cancellation if some platform uses up_tls_info. nxnotify_cancellation is in the kernel (for now) and it needs to find the TLS area of another thread, I made tls_get_info_pid() for this. up_tls_info can only locate the TLS of the current thread.

Another option would be to move nxnotify_cancellation to libc as well, and just sending a signal to the canceled thread instead. This will require moving pthread_cancel to libc though.